internal/controller/isbservicerollout_controller.go

```diff
@@ -215,14 +217,18 @@ func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceR
 		Spec: isbServiceRollout.Spec.InterStepBufferService.Spec,
 	}
 
-	existingISBServiceDef, err := kubernetes.GetCR(ctx, r.restConfig, newISBServiceDef, "interstepbufferservices")
+	existingISBServiceDefUns, err := kubernetes.ObjectToUnstructured(newISBServiceDef)
```

Comment:
"existing" should only mean "isbsvc that exists in K8S". "new" should only mean the new one which is defined by the ISBServiceRollout